File: fadl_pos/fadl_pos/report/current_stock_report/current_stock_report.py
# Copyright (c) 2026, FadlTech team and contributors

from __future__ import annotations



import frappe
from frappe import _


def execute(filters=None):
	filters = frappe._dict(filters or {})
	data = get_filtered_items(filters)
	return get_columns(), data


def get_filtered_items(filters):
	query = """
        SELECT
            ti.item_code,
            ti.item_name,
            ti.brand,
            ti.item_group,
            bin.actual_qty AS stock,
            bin.valuation_rate,
            bin.stock_value
        FROM `tabItem` ti
        INNER JOIN `tabBin` bin ON ti.item_code = bin.item_code
        WHERE bin.actual_qty > 0
            AND bin.warehouse = %(warehouse)s
    """

	conditions = {}
	conditions["warehouse"] = filters.get("warehouse")

	if filters.get("supplier"):
		query += """
            AND ti.item_code IN (
                SELECT parent FROM `tabItem Supplier` WHERE supplier = %(supplier)s
            )
        """
		conditions["supplier"] = filters.get("supplier")

	if filters.get("brand"):
		query += " AND ti.brand = %(brand)s"
		conditions["brand"] = filters.get("brand")

	if filters.get("item_group"):
		query += " AND ti.item_group = %(item_group)s"
		conditions["item_group"] = filters.get("item_group")

	query += " ORDER BY ti.item_name"

	return frappe.db.sql(query, conditions, as_dict=True)


def get_columns():
	return [
		{
			"label": _("Item Code"),
			"fieldname": "item_code",
			"fieldtype": "Link",
			"options": "Item",
			"width": 150,
		},
		{
			"label": _("Item Name"),
			"fieldname": "item_name",
			"fieldtype": "Data",
			"width": 200,
		},
		{
			"label": _("Item Group"),
			"fieldname": "item_group",
			"fieldtype": "Data",
			"width": 200,
		},
		{
			"label": _("Brand"),
			"fieldname": "brand",
			"fieldtype": "Link",
			"options": "Brand",
			"width": 120,
		},
		{"label": _("Stock"), "fieldname": "stock", "fieldtype": "Float", "width": 100},
		{
			"label": _("Valuation Rate"),
			"fieldname": "valuation_rate",
			"fieldtype": "Currency",
			"width": 120,
		},
		{
			"label": _("Stock Value"),
			"fieldname": "stock_value",
			"fieldtype": "Currency",
			"width": 120,
		},
	]

File: fadl_pos/fadl_pos/report/current_stock_report/current_stock_report.js
// Copyright (c) 2025, Ali Raza and contributors
// For license information, please see license.txt

frappe.query_reports["Current Stock Report"] = {
	filters: [
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			reqd: 1,
			default: frappe.defaults.get_user_default("Company"),
			on_change: function () {
				frappe.query_report.set_filter_value("warehouse", null);
			},
		},
		{
			fieldname: "warehouse",
			label: __("Warehouse"),
			fieldtype: "Link",
			options: "Warehouse",
			reqd: 1,
			get_query: () => {
				return {
					filters: {
						company: frappe.query_report.get_filter_value("company"),
						is_group: 0,
					},
				};
			},
		},
		{
			fieldname: "supplier",
			label: __("Supplier"),
			fieldtype: "Link",
			options: "Supplier",
		},
		{
			fieldname: "brand",
			label: __("Brand"),
			fieldtype: "Link",
			options: "Brand",
		},
		{
			fieldname: "item_group",
			label: __("Item Group"),
			fieldtype: "Link",
			options: "Item Group",
		},
	],
};
